src/main/java/com/ldtteam/blockout/hooks/HookManager.java

```diff
@@ -0,0 +1,235 @@
+package com.ldtteam.blockout.hooks;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import com.ldtteam.blockout.hooks.TriggerMechanism.Type;
+import com.ldtteam.structurize.api.util.constant.Constants;
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+
+public abstract class HookManager<T, U extends IForgeRegistryEntry<U>, K>
```

Comment:
this should also have an interface with javadoc for the methods

Reply:
you mean something like IHookManager? why? this is not part of visible API imho

Reply:
Mostly to have a nice place for all the docu

Reply:
Like there is nothing to document, can document registerInternal but that's all, tick and render are quite obvious